### PR TITLE
SpongeForge for Minecraft 1.20.6

### PR DESCRIPTION
**Pull request by Yeregorix**

### CODE REVIEW

File: forge/src/mixins/java/org/spongepowered/forge/mixin/core/server/network/ServerLoginPacketListenerImplMixin_Forge.java [hunk 1]
```diff
@@ -24,51 +24,27 @@
...
-        }
+    @Shadow private ServerLoginPacketListenerImpl.State state;
+    // @formatter:on
 
-        if (this.bridge$fireAuthEvent()) {
-            ci.cancel();
+    @Inject(method = "tick", at = @At("HEAD"))
+    private void impl$onTick(final CallbackInfo ci) {
+        // In SpongeVanilla we do channel registration during this state, not sure if we need to do anything in SpongeForge
+        if (this.state == ServerLoginPacketListenerImpl.State.NEGOTIATING) {
```
Comment (Zidane) [inline forge/src/mixins/java/org/spongepowered/forge/mixin/core/server/network/ServerLoginPacketListenerImplMixin_Forge.java:46]: Why does this change exist?

File: src/mixins/java/org/spongepowered/common/mixin/core/network/ConnectionMixin.java [hunk 2]
```diff
@@ -156,13 +155,6 @@ public abstract class ConnectionMixin extends SimpleChannelInboundHandler<Packet
         this.impl$version = new SpongeMinecraftVersion(String.valueOf(version), version);
     }
 
-    @Inject(method = "lambda$doSendPacket$10", at = @At(value = "INVOKE", target = "Lio/netty/util/concurrent/Future;isSuccess()Z"))
```
Comment (Zidane) [inline src/mixins/java/org/spongepowered/common/mixin/core/network/ConnectionMixin.java]: Why was this removed?

File: forge/src/mixins/java/org/spongepowered/forge/mixin/core/server/network/ServerLoginPacketListenerImplMixin_Forge.java (hunk 1 above)
Comment (Yeregorix) [reply forge/src/mixins/java/org/spongepowered/forge/mixin/core/server/network/ServerLoginPacketListenerImplMixin_Forge.java:46]: See https://github.com/SpongePowered/Sponge/blob/api11/forge/src/mixins/java/org/spongepowered/common/mixin/core/server/network/ServerLoginPacketListenerImplMixin.java#L146
and https://github.com/SpongePowered/Sponge/blob/api11/forge/vanilla/src/mixins/java/org/spongepowered/vanilla/mixin/core/server/network/ServerLoginPacketListenerImplMixin_Vanilla.java#L80

After auth, we switch to state `NEGOTIATING` which is unused by vanilla MC. We use this state in SpongeVanilla to register channels and then we go to `VERIFYING` which will be handled by vanilla MC.

I had to replicate the `NEGOTIATING` -> `VERIFYING` switch in SpongeForge otherwise the login stay forever in state `NEGOTIATING`.

File: src/mixins/java/org/spongepowered/common/mixin/core/network/ConnectionMixin.java (hunk 2 above)
Comment (Yeregorix) [reply src/mixins/java/org/spongepowered/common/mixin/core/network/ConnectionMixin.java]: Moved to `ConnectionMixin_Vanilla` and `ConnectionMixin_Forge` due to injection error.
